platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,163 @@
+
+    def __exit__(self, *args):
+        self.file_obj.close()
+
+
+class DpuCtlPlat():
+    """Class for Per DPU API Call"""
+    def __init__(self, dpu_name):
+        self._name = dpu_name
+        self.set_go_down_path = os.path.join(SYSTEM_BASE,
+                                             f"{self._name}_rst")
+        self.set_pwr_path = os.path.join(SYSTEM_BASE,
+                                         f"{self._name}_pwr")
+        self.set_pwr_f_path = os.path.join(SYSTEM_BASE,
+                                           f"{self._name}_pwr_force")
+        self.get_dpu_rdy_path = os.path.join(EVENT_BASE,
+                                             f"{self._name}_ready")
+        self.set_dpu_perst_en_path = os.path.join(SYSTEM_BASE,
+                                                  f"{self._name}_perst_en")
+
+    def write_file(self, file_name, content_towrite):
+        """Write given value to file only if file exists"""
+        try:
+            with DataWriter(file_name) as file_obj:
+                file_obj.write(content_towrite)
+        except Exception as file_write_exc:
+            logger.log_error(f'{self.get_name()}:Failed to write'
+                             f'{content_towrite} to file {file_name}')
+            raise type(file_write_exc)(
+                f"{self.get_name()}:{str(file_write_exc)}")
+        return True
+
+    def get_name(self):
+        """Return name of the DPU"""
+        return self._name
+
+    def dpu_go_down(self):
+        """Per DPU going down API"""
+        get_shtdn_ready_path = os.path.join(EVENT_BASE,
+                                            f"{self.get_name()}_shtdn_ready")
+        try:
+            get_shtdn_inotify = InotifyHelper(get_shtdn_ready_path)
+            dpu_shtdn_rdy = get_shtdn_inotify.add_watch(WAIT_FOR_SHTDN, 1)
+        except (FileNotFoundError, PermissionError) as inotify_exc:
+            raise type(inotify_exc)(f"{self.get_name()}:{str(inotify_exc)}")
+        if dpu_shtdn_rdy is None:
+            logger.log_info(f"{self.get_name()}: Going Down Unsuccessful")
+            self.dpu_power_off(forced=True)
+            return None
+        return True
+
+    def dpu_power_off(self, forced=False):
+        """Per DPU Power off API"""
+        logger.log_info(f"{self.get_name()}: Power off forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "1")
+            logger.log_info(f"{self.get_name()}: Force Power Off complete")
+            return True
+        self.write_file(self.set_go_down_path, "1")
+        if not self.dpu_go_down(): 
+            """If go_down returns None, 
+            that means Force power off is executed, No need to power off"""
+            return True
+        self.write_file(self.set_pwr_path, "1")
+        logger.log_info(f"{self.get_name()}: Power Off complete")
+        return True
+
+    def _power_on_force(self, count=4):
+        """Per DPU Power on with force private function"""
```

Comment:
Shouldn't the dpu_power_off(force=True) be called from here?

Reply:
Done